app/main/views/root.py

```diff
@@ -75,7 +75,14 @@ def login():
 
     questionnaire_manager = QuestionnaireManagerFactory.get_instance()
 
-    # get the current location of the user
-    current_location = questionnaire_manager.get_current_location()
+    current_location = questionnaire_manager.navigator.get_first_location()
+
+    visited_blocks = get_visited_blocks(current_user)
+
+    if visited_blocks:
+        incomplete_blocks = [item for item in questionnaire_manager.navigator.get_location_path(get_answers(current_user)) if item not in visited_blocks]
+
+        if incomplete_blocks:
+            current_location = incomplete_blocks[0]
```

Comment:
It is replaceable, i must have missed this when i moved the logic into get_current_location